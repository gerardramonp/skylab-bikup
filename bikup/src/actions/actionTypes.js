export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
