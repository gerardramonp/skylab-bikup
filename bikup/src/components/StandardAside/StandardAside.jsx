import React, { useState, useEffect } from 'react';
import './StandardAside.scss';

function StandardAside() {
    const challenges = [
        {
            name: 'BIrlfriend',
            description: 'Travel 5000km with a bike'
        },
        {
            name: 'CompoManiac',
            description: 'Change the same component 3 times'
        }
    ];

    return (
        <div className="standardaside">
            <h3>Challenges</h3>
            <div className="standardaside__card">
                {challenges.map((challenge) => {
                    return (
                        <div>
                            <div className="challenges__item">
                                <div className="item__title">
                                    <img
                                        className="title__img"
                                        src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                                        alt="trophy"
                                    />
                                    - {challenge.name} -
                                </div>
                                <p>{challenge.description}</p>
                            </div>
                            <div className="separator"></div>
                        </div>
                    );
                })}
            </div>

            <h3>Give Away</h3>
            <div className="standardaside__card">
                <p>something free</p>
            </div>
        </div>
    );
}

export default StandardAside;
