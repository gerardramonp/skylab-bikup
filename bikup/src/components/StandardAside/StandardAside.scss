$light-bg-color: #eeeeee;
$light-highlight-color: #d7d7d7;
$main-color: #32e0c4;

.standardaside {
    display: none;
    padding: 12px;
}

.standardaside__card {
    padding: 12px;
    background-color: $light-highlight-color;
    border-radius: 5px;
    margin-top: 10px;
    margin-bottom: 64px;

    box-shadow: 5px 5px 5px 0px rgba(209, 205, 209, 1);
}

.challenges__item {
    margin-bottom: 10px;
}

.item__title {
    font-weight: bold;
    display: flex;
    align-items: center;
    margin-bottom: 10px;
}

.title__img {
    height: 18px;
    margin-right: 10px;
}

.separator {
    width: 90%;
    margin: auto;
    border-bottom: 0.5px solid grey;
    margin-bottom: 12px;
}

@media only screen and (min-width: 600px) {
    .standardaside {
        display: flex;
        flex-direction: column;
    }
}
