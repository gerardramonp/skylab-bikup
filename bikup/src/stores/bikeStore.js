import dispatcher from '../dispatcher';
import actionTypes from '../actions/actionTypes';
import { EventEmitter } from 'events';

const CHANGE_EVENT = 'change';

let _bike = {};

class BikeStore extends EventEmitter {
	addChangeListener(callback) {
		this.on(CHANGE_EVENT, callback);
	}

	removeChangeListener(callback) {
		this.removeListener(CHANGE_EVENT, callback);
	}

	emitChange() {
		this.emit(CHANGE_EVENT);
	}

	setBikeDetail(bikeDetail) {
		_bike = bikeDetail;
	}

	getBikeDetail() {
		return _bike;
	}
}

const bikeStore = new BikeStore();

dispatcher.register((action) => {
	switch (action.type) {
		case actionTypes.LOAD_BIKE_BY_ID:
			bikeStore.setBikeDetail(action.data);
			bikeStore.emitChange();
			break;
		default:
			console.log(`There is no action with type: ${action.type}`);
			break;
	}
});

export default bikeStore;
