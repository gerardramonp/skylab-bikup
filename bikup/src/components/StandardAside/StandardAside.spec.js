import React from 'react';
import renderer from 'react-test-renderer';
import StandardAside from './StandardAside';
import { BrowserRouter as Router } from 'react-router-dom';
import dispatcher from '../../dispatcher';
import bikeStore from '../../stores/bikeStore';
import actionTypes from '../../actions/actionTypes';

describe('Standard Aside', () => {
    let tree = renderer.create(
        <Router>
            <StandardAside />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
