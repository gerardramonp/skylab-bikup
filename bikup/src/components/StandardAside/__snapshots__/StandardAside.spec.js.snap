// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standard Aside should match 1`] = `
<div
  className="standardaside"
>
  <div
    className="standardaside__card-container"
  >
    <h3
      className="title"
    >
      Challenges
    </h3>
    <div
      className="standardaside__card"
    >
      <div>
        <div
          className="challenges__item"
        >
          <div
            className="item__title"
          >
            <img
              alt="trophy"
              className="title__img"
              src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
            />
            - 
            BIrlfriend
             -
          </div>
          <p>
            Travel 5000km with a bike
          </p>
        </div>
        <div
          className="separator"
        />
      </div>
      <div>
        <div
          className="challenges__item"
        >
          <div
            className="item__title"
          >
            <img
              alt="trophy"
              className="title__img"
              src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
            />
            - 
            CompoManiac
             -
          </div>
          <p>
            Change the same component 3 times
          </p>
        </div>
        <div
          className="separator"
        />
      </div>
    </div>
  </div>
  <div
    className="standardaside__card-container"
  >
    <h3
      className="title"
    >
      Give Away
    </h3>
    <div
      className="standardaside__card"
    >
      <p>
        something free
      </p>
    </div>
  </div>
</div>
`;
